### PR TITLE
Add "supportForTags" field to tsdoc.json schema

### PR DESCRIPTION
**Pull request by octogonz**
This PR also adds the ability to generate a **tsdoc.json** file containing the `TSDocConfiguration` state.

### CODE REVIEW

File: tsdoc-config/src/TSDocConfigFile.ts [hunk 1]
```diff
@@ -132,6 +138,75 @@ export class TSDocConfigFile {
...
 
+  public get supportForTags(): ReadonlyMap<string, boolean> {
+    return this._supportForTags;
+  }
+
+  /**
+   * Removes all items from the `tagDefinitions` array.
+   */
+  public clearTagDefinitions(): void {
+    this._tagDefinitions.length = 0;
```
Comment (iclanton) [inline tsdoc-config/src/TSDocConfigFile.ts:149]: Does this really clear the array? Why not just

```suggestion
    this._tagDefinitions = [];
```

File: tsdoc-config/src/TSDocConfigFile.ts [hunk 2]
```diff
@@ -7,12 +7,14 @@ import {
...
   TextRange,
   IParserMessageParameters,
+  ITSDocTagDefinitionParameters,
 } from '@microsoft/tsdoc';
 import * as fs from 'fs';
 import * as resolve from 'resolve';
 import * as path from 'path';
 import Ajv from 'ajv';
 import * as jju from 'jju';
+import { config } from 'process';
```
Comment (iclanton) [inline tsdoc-config/src/TSDocConfigFile.ts:17]: Unused?

File: tsdoc/src/configuration/TSDocConfiguration.ts [hunk 3]
```diff
@@ -22,13 +22,30 @@ export class TSDocConfiguration {
...
     BuiltInDocNodes.register(this);
   }
 
+  /**
+   * Resets the `TSDocConfiguration` object to its initial empty state.
+   * @param noStandardTags - The `TSDocConfiguration` constructor normally adds definitions for the
+   * standard TSDoc tags.  Set `noStandardTags` to true for a completely empty `tagDefinitions` collection.
+   */
+  public clear(noStandardTags: boolean = false): void {
+    this._tagDefinitions.length = 0;
```
Comment (iclanton) [inline tsdoc/src/configuration/TSDocConfiguration.ts:37]:
```suggestion
    this._tagDefinitions = [];
```

**Review by iclanton — APPROVED** [review]

File: tsdoc-config/src/TSDocConfigFile.ts (hunk 1 above)
Comment (octogonz) [reply tsdoc-config/src/TSDocConfigFile.ts:149]: It's the only way to clear a `readonly` array.  I think this is a better practice than replacing the object with `[ ]`.

File: tsdoc/src/configuration/TSDocConfiguration.ts (hunk 3 above)
Comment (octogonz) [reply tsdoc/src/configuration/TSDocConfiguration.ts:37]: See other reply

**Pull request merged**